__tests__/app.test.js

```diff
@@ -25,30 +27,75 @@ describe("/api", ()=>{
     })
 })
 
-describe("/api/anything", () => {
+describe("General error handling", () => {
     test("GET: 404 - responds with msg: End point not found", () => {
         return request(app)
         .get("/anything")
         .expect(404)
         .then(({ body }) => {
             expect(body.msg).toBe("End point not found");
         });
+    });
 });
- 
   
 describe("/api/topics", () => {
     test("GET:200 - responds with all topics", () => {
         return request(app)
         .get("/api/topics")
         .expect(200)
         .then(({ body }) => {
+          expect(body.topics).toHaveLength(3);
           body.topics.forEach((topic) => {
-            expect(Object.keys(topic).length).toBe(2);
+            expect(Object.keys(topic).length).toBeGreaterThan(0);
             expect(typeof topic.description).toBe("string");
             expect(typeof topic.slug).toBe("string");
           });
         });
-    });
-    
+    });   
+});
+
+describe("/api/articles/:article_id", () => {
+  test("GET:200 - responds with the article matching the article id", () => {
+    return request(app)
+      .get("/api/articles/4")
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.article).toMatchObject({
```

Comment:
This test is fine, but as we know the shape of the article we are receiving, we could also have asserted against the actual values we get back